### DISCUSSION TITLE
Get SenderId mismatch error while trying to create notification

### DISCUSSION THREAD

**Discussion by nakarimi** [kreait/firebase-php#610] [General]
I checked the project number and sender id that are the same. Server key also is checked and updated.
the cloud api also is enabled https://console.cloud.google.com/api.

what could make this error?

here is the code
`        $title = 'My Notification Title';
        $body = 'My Notification Body';
        $notification = Notification::create($title, $body);

        $path = base_path('config/fooddelivery-firebase.json');
        $factory = (new Factory)
            ->withServiceAccount($path)
            ->withDatabaseUri('https://fooddelivery-cc39b-default-rtdb.firebaseio.com');
            
        // Send notification by token.
        $messaging = $factory->createMessaging();                
        $deviceToken = 'AAAAfddQBiE:APA91bHKAgwHn4jKvd...EOXEBqpfExUGLu_7XNSbkY';
        $message = CloudMessage::withTarget('token', $deviceToken)
            ->withNotification($notification);
        return $messaging->send($message);
`

and error : 
SenderId mismatch in file /home/pc/Desktop/new PC/devfood/vendor/kreait/firebase-php/src/Firebase/Exception/Messaging/AuthenticationError.php on line 22

**Comment by jeromegamez** [comment]
Have you also ensured that the service account and the registration token belong to the same project?

The code looks fine, but I want to add that the Sender ID and Project ID are two different things and a Sender ID is not used within the SDK. It's only the service account's project ID that matters.

**Comment by nakarimi** [comment]
Does the registration token is same as server key that could be found on https://console.firebase.google.com/u/1/project/fooddelivery-cc39b/settings/cloudmessaging/ link?

This is the service account data 
`redacted`
and I got the registration token or server key from this link https://console.firebase.google.com/u/1/project/redacted/settings/cloudmessaging/, so project id matched.

**Reply by jeromegamez** [reply]
Server Key != Registration Token

You don't need a Server Key when using the Admin SDK.

The registration token is a unique ID that identifies a client device (e.g. a web app or a mobile phone). The official docs explain what a registration token is in detail, and how you can get access to it, for example in a web application: https://firebase.google.com/docs/cloud-messaging/js/first-message#access_the_registration_token, or on Android: https://firebase.google.com/docs/cloud-messaging/android/first-message#access_the_registration_token). You cannot access a registration token from the Firebase Web Console.

The server (Admin SDK) and your client (Client SDK) must be configured with the same Firebase project - you cannot send messages between two separate Firebase Projects.

Another reason could be that the registration token is outdated/has been revoked - in this case, please try with a newly issued registration token. 🤞
